### PR TITLE
Log info about precompiled letter software

### PR DESCRIPTION
**Pull request by benthorner**
https://www.pivotaltracker.com/story/show/177985853

See the comment for more details. If you're wondering what the difference 
between "creator" and "producer" is:

- The "creator" is the original publishing software.
- The "producer" is the software used to convert the original format to a PDF (if any).

### CODE REVIEW

**Review by CrystalPea — APPROVED** [review]

**Comment by benthorner** [comment]
> are they ever not present? (and will they ever AttributeError?)

@leohemsted raised a good point. Although the individual attributes will be fine, it's also possible for a PDF not to have an `/Info` section. I've added a new commit to handle that, with a test to cover it properly.

@CrystalPea do you want to take another look?

File: tests/test_precompiled.py [hunk 1]
```diff
@@ -558,6 +561,20 @@ def test_sanitise_precompiled_letter_with_bad_address_returns_400(
...
+    pdf_with_no_metadata,
+], ids=['valid_letter', 'pdf_with_no_metadata'])
+def test_log_metadata_for_letter(
+    client,
+    file,
+    mocker,
+):
+    logger = mocker.patch('app.precompiled.current_app.logger.info')
+    log_metadata_for_letter(BytesIO(file), 'filename')
+    assert logger.called
```
Comment (CrystalPea) [inline tests/test_precompiled.py:575]: would we like to assert that it's called with a correct message?

Comment (benthorner) [reply tests/test_precompiled.py:575]: That would be more effort, but the log isn't critical: just for analysis.

I've tested it works manually, so I'm satisfied it's been sufficiently tested for the time being.

If it does break in future, we can always make the test more prescriptive.

Comment (CrystalPea) [reply tests/test_precompiled.py:575]: ok 👍🏼

**Review by CrystalPea — APPROVED** [review]

**Pull request merged**